lib/modules/datasource/docker/index.ts

```diff
@@ -812,54 +812,144 @@ export class DockerDatasource extends Datasource {
   @cache({
     namespace: 'datasource-docker-digest',
     key: (
-      { registryUrl, packageName }: GetReleasesConfig,
+      { registryUrl, packageName, currentDigest }: DigestConfig,
       newValue?: string
     ) => {
       const newTag = newValue ?? 'latest';
       const { registryHost, dockerRepository } = getRegistryRepository(
         packageName,
         registryUrl!
       );
-      return `${registryHost}:${dockerRepository}:${newTag}`;
+      const digest = currentDigest ? '@' + currentDigest : '';
+      return `${registryHost}:${dockerRepository}:${newTag}${digest}`;
     },
   })
   override async getDigest(
-    { registryUrl, packageName }: GetReleasesConfig,
+    { registryUrl, packageName, currentValue, currentDigest }: DigestConfig,
     newValue?: string
   ): Promise<string | null> {
     const { registryHost, dockerRepository } = getRegistryRepository(
       packageName,
       registryUrl!
     );
     logger.debug(
-      `getDigest(${registryHost}, ${dockerRepository}, ${newValue})`
+      `getDigest(${registryHost}, ${dockerRepository}, ${
+        newValue ?? 'undefined'
+      })`
     );
     const newTag = newValue ?? 'latest';
     let digest: string | null = null;
     try {
-      let manifestResponse = await this.getManifestResponse(
-        registryHost,
-        dockerRepository,
-        newTag,
-        'head'
-      );
-      if (manifestResponse) {
-        if (hasKey('docker-content-digest', manifestResponse.headers)) {
-          digest =
-            (manifestResponse.headers['docker-content-digest'] as string) ||
-            null;
-        } else {
-          logger.debug(
-            { registryHost },
-            'Missing docker content digest header, pulling full manifest'
-          );
+      let architecture: string | null = null;
+      if (currentValue && currentDigest) {
+        let manifestResponse = await this.getManifestResponse(
+          registryHost,
+          dockerRepository,
+          currentDigest,
+          'head'
+        );
+
+        if (
+          manifestResponse &&
+          (manifestResponse.headers['content-type'] === MediaType.manifestV2 ||
+            manifestResponse.headers['content-type'] ===
+              MediaType.ociManifestV1)
+        ) {
           manifestResponse = await this.getManifestResponse(
             registryHost,
             dockerRepository,
-            newTag
+            currentValue
           );
+          if (manifestResponse) {
+            const manifestList = JSON.parse(manifestResponse.body) as
+              | ImageList
+              | Image
+              | OciImageList
+              | OciImage;
```

Comment:
```suggestion
              | Image
              | OciImage;
```
should only be an image manifest, it should also have an architecture

Reply:
The image manifest doesn't have an architecture, see [here](https://docs.docker.com/registry/spec/manifest-v2-2/#example-image-manifest). This is why I initially only had manifest list:
```ts
| ImageList
| OciImageList;
```
but in https://github.com/renovatebot/renovate/pull/16554#discussion_r919673465, you requested to add 
```ts
| Image
| OciImage;
```